nemo/collections/asr/metrics/wer.py

```diff
@@ -148,12 +148,16 @@ def ctc_decoder_predictions_tensor(
         """
         hypotheses = []
         # Drop predictions to CPU
-        predictions = move_dimension_to_the_front(predictions, self.batch_dim_index)
-        prediction_cpu_tensor = predictions.long().cpu()
+        if isinstance(predictions, torch.Tensor):
+            predictions = move_dimension_to_the_front(predictions, self.batch_dim_index)
+            prediction_cpu_tensor = predictions.long().cpu()
+        else:
+            prediction_cpu_tensor = predictions
+
         # iterate over batch
-        for ind in range(prediction_cpu_tensor.shape[0]):
+        for ind in range(len(prediction_cpu_tensor)):
             if self.fold_consecutive:
-                prediction = prediction_cpu_tensor[ind].detach().numpy().tolist()
+                prediction = prediction_cpu_tensor[ind].cpu().detach().numpy().tolist()
```

Comment:
This.. is very expensive. Leave the tensor to its device otherwise it is very costly as its done in a loop.

Reply:
dropped it.